tests/binary-cache-sodium.sh

```diff
@@ -0,0 +1,69 @@
+export NIX_TEST_ROOT="$(cd -P -- "$(dirname -- "$0")" && pwd -P)"
+source "$NIX_TEST_ROOT/common.sh"
+
+setupTest
+
+cacheDir=$TEST_ROOT/binary-cache
+
+clearCache() {
+    rm -rf "$cacheDir"
+}
+
+clearCacheCache() {
+    rm -f $TEST_HOME/.cache/nix/binary-cache*
+}
+
+# Create a signed binary cache.
+outPath=$(nix-build $NIX_TEST_ROOT/dependencies.nix --no-out-link)
```

Comment:
This script will not be executed if nix wasn't configured with sodium, but we don't have access to a variable to tell us so (and that is why this script was split into a separate file).